packages/smooth_app/lib/pages/product/product_image_viewer.dart

```diff
@@ -0,0 +1,134 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:http/http.dart' as http;
+import 'package:path_provider/path_provider.dart';
+import 'package:photo_view/photo_view.dart';
+import 'package:smooth_app/data_models/product_image_data.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/generic_lib/loading_dialog.dart';
+import 'package:smooth_app/generic_lib/widgets/smooth_back_button.dart';
+import 'package:smooth_app/pages/product/confirm_and_upload_picture.dart';
+import 'package:smooth_app/widgets/smooth_scaffold.dart';
+
+/// Displays a full-screen image with an edit floating button
+class ProductImageViewer extends StatefulWidget {
+  const ProductImageViewer({
+    required this.barcode,
+    required this.imageData,
+  });
+
+  final String barcode;
+  final ProductImageData imageData;
+
+  @override
+  State<ProductImageViewer> createState() => _ProductImageViewerState();
+}
+
+class _ProductImageViewerState extends State<ProductImageViewer> {
+  late final ProductImageData imageData;
+
+  /// When the image is edited, this is the new image
+  ImageProvider? imageProvider;
+  bool _isEdited = false;
+
+  @override
+  void initState() {
+    imageData = widget.imageData;
+
+    if (imageData.imageUrl != null) {
+      imageProvider = NetworkImage(imageData.imageUrl!);
+    }
+
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) => SmoothScaffold(
+        extendBodyBehindAppBar: true,
+        backgroundColor: Colors.black,
+        floatingActionButton: createEditButton(onPressed: _editImage),
+        appBar: AppBar(
+          backgroundColor: Colors.black,
+          foregroundColor: WHITE_COLOR,
+          elevation: 0,
+          title: Text(imageData.title),
+          leading: SmoothBackButton(
+            onPressed: () => Navigator.maybePop(context, _isEdited),
+          ),
+        ),
+        body: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            ConstrainedBox(
+              constraints: BoxConstraints.tight(
+                Size(double.infinity, MediaQuery.of(context).size.height / 2),
+              ),
+              child: PhotoView(
+                minScale: 0.2,
+                imageProvider: imageProvider,
+                heroAttributes: PhotoViewHeroAttributes(
+                  tag: imageProvider!,
```

Comment:
What do you mean, you cannot guarantee `imageProvider` is not null? You'd better!

Reply:
Right, removing the check!